pkg/ui/workspaces/cluster-ui/src/util/hooks.ts

```diff
@@ -193,3 +194,28 @@ export const useSwrMutationWithClusterId = <
   const keyWithClusterId = useSwrKeyWithClusterId(key) as SWRKey;
   return useSWRMutation(keyWithClusterId, fetcher, config);
 };
+
+export const usePagination = (
+  defaultPage: number,
+  defaultPageSize: number,
+): [
+  ISortedTablePagination,
+  (current: number, pageSize: number) => void,
+  () => void,
+] => {
+  const [pagination, setPagination] = useState<ISortedTablePagination>({
+    current: defaultPage,
+    pageSize: defaultPageSize,
+  });
+  const updatePagination = (current: number, pageSize: number) => {
+    setPagination({
+      ...pagination,
```

Comment:
This spread seems redundant since we're writing all the object fields below.

Reply:
Removed